core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -3435,6 +3436,14 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
         maybeUpdateGoogleProjectsInPerimeter(billingProject, dataAccess, context.tracingSpan.orNull)
       )
 
+      // If an Azure workspace -- After the workspace has been created, create a WDS instance via Leonardo corresponding with the workspace
+      // TODO: determine how to determine if Azure workspace if (isAzureMcWorkspace(maybeLoadMcWorkspace(savedWorkspace))) (
+      _ = logger.info(s"creating WDS instance - workspace:'${workspaceName}' - UUID:${workspaceId}")
+      _ <- traceWithParent("createWDSInstance", parentContext) { context =>
+        val leonardoClient = new LeonardoClient("a base path");
+        leonardoClient.createWDSInstance("a token", workspaceId, "workspace Name that should be casted to String");
+      }
+
```

Comment:
this may be moot because the code needs to move, but there's a few suggestions here:
* suggest moving the logging statement inside the closure instead of outside
* the `for … yield` block containing this new code is a series of `DBIOAction`s, so this code also needs to be a `DBIOAction`:
  * use `traceDBIOWithParent` instead of `traceWithParent`
  * `traceDBIOWithParent` requires its inner code to return a DBIO, which you can create with `DBIO.from()`
  * `DBIO.from` requires its inner code to be a Future, which you can create with `Future`
* the "context" variable returned from `traceWithParent`/`traceDBIOWithParent` is totally unused; you can tell the compiler to ignore this by naming it `_`.
* you can get the current user's access token from `parentContext.userInfo` as the first argument to leonardoClient.createWDSInstance
* you can use `s"…"` string interpolation to set the app name as the third argument to `leonardoClient.createWDSInstance`


```suggestion
      _ <- traceDBIOWithParent("createWDSInstance", parentContext) { _ =>
        logger.info(s"creating WDS instance - workspace:'${workspaceName}' - UUID:${workspaceId}")
        val leonardoClient = new LeonardoClient("a base path")
        DBIO.from(Future(leonardoClient.createWDSInstance(parentContext.userInfo.accessToken.token,
          workspaceId, s"wds-${workspaceId}")))
      }
```

Reply:
Updated to `traceWithParent` where workspace creation occurs in Rawls